mods/bucket/init.lua

```diff
@@ -76,24 +76,28 @@ function bucket.register_liquid(source, flowing, itemname, inventory_image, name
 					if check_protection(pos,
 							user and user:get_player_name() or "",
 							"place "..source) then
-						return
+						return nil
```

Comment:
This is in an `on_place()` callback. It should return `itemstack` unmodified if the placement fails.

Because all you're doing is `if place_liquid() return...` then might as well write it this way:

```
return place_liquid(pos, itemstack, node, source, flowing)
```

and have place_liquid return the right thing:

```
return itemstack
```

place_liquid should really be defined *outside* this scope so we're not redefining it 3x.